ci-operator/step-registry/aws/provision/bastionhost/proxy-whitelist/aws-provision-bastionhost-proxy-whitelist-commands.sh

```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+REGION=${LEASED_RESOURCE}
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+function run_command() {
+    local cmd="$1"
+    echo "Running Command: ${cmd}"
+    eval "${cmd}"
+}
+
+function run_ssh_cmd() {
+    local sshkey=$1
+    local user=$2
+    local host=$3
+    local remote_cmd=$4
+    local options=$5
+    local cmd
+    options+=" -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o VerifyHostKeyDNS=yes "
+    cmd="ssh ${options} -i \"${sshkey}\" ${user}@${host} \"${remote_cmd}\""
+    run_command "$cmd" || return 2
+    return 0
+}
+
+function run_scp_to_remote() {
+    local sshkey=$1
+    local user=$2
+    local host=$3
+    local src=$4
+    local dest=$5
+    local options=$6
+    local cmd
+    options+=" -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o VerifyHostKeyDNS=yes "
+    cmd="scp ${options} -i \"${sshkey}\" ${src} ${user}@${host}:${dest}"
+    run_command "$cmd" || return 2
+    return 0
+}
+
+proxy_host_address=$(head -n 1 "${SHARED_DIR}/bastion_public_address")
+proxy_host_user=$(head -n 1 "${SHARED_DIR}/bastion_ssh_user")
+
+
+# https://docs.openshift.com/container-platform/4.15/installing/install_config/configuring-firewall.html#configuring-firewall
+
+# Registry URLs
+cat <<EOF > ${ARTIFACT_DIR}/allowlist.txt
+access.redhat.com
+cdn.quay.io
+cdn01.quay.io
+cdn02.quay.io
+cdn03.quay.io
+quay.io
+registry.redhat.io
+sso.redhat.com
+EOF
+
+# Telemetry
+cat <<EOF >> ${ARTIFACT_DIR}/allowlist.txt
+api.access.redhat.com
+cert-api.access.redhat.com
+console.redhat.com
+infogw.api.openshift.com
+EOF
```

Comment:
I am thinking if it deserved to extract the above url into a separated step so that other platform can share it.

Reply:
we could, but need to add a platform checking condition, aws, gcp, azure

Reply:
For "Registry URLs" and "Telemetry", they are different on different platforms?

Reply:
No, they are for all platforms.

Reply:
Or rename the step name to something like `enable-proxy-whitelist`, all URLs are maintained in this step and can be invoked by all platforms.

Reply:
I meant only extract the step of writing  "Registry URLs" and "Telemetry" in a separate step, then for for the different urls on different platforms, each platform can write their own step, something like:
step-1: write platform dependent urls into allowlist.txt
step-2: append common urls into allowlist.txt, then run ssh command to re-configure proxy.

Reply:
like this: 
```
chain:
  as: proxy-whitelist-aws
  steps:
    - ref: proxy-whitelist-aws
    - ref: proxy-whitelist-enable
```

Reply:
right.